MCT configure error with IBM compilers

MCT used to build fine with a previous version of XL compilers. It can't get past the configuration stage. 

```
gmake -f /autofs/nccs-svm1_home1/sarat/repos/E3SM/cime/scripts/fc5_ibm/Tools/Makefile  -C /gpfs/alpinetds/scratch/sarat/csc190/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct  CASEROOT=/autofs/nccs-svm1_home1/sarat/repos/E3SM/cime/scripts/fc5_ibm MODEL=mct /gpfs/alpinetds/scratch/sarat/csc190/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct/Makefile.conf
gmake: Entering directory `/gpfs/alpinetds/csc190/scratch/sarat/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct'
...
configure: error: in `/gpfs/alpinetds/csc190/scratch/sarat/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct':
configure: error: C compiler cannot create executables
See `config.log' for more details
gmake: *** [/gpfs/alpinetds/scratch/sarat/csc190/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct/Makefile.conf] Error 77
```

config.log
```
exec: export(export,XL_CONFIG=/sw/summit/xl/20180502/xlC/16.1.0/etc/xlc.cfg.rhel.7.5.gcc.4.8.5.cuda.9.2:xlc,NULL) 
unlink: /tmp/xlcW0TCnskh
unlink: /tmp/xlcW17rY04M
unlink: /tmp/xlcW2vIyzPi
/sw/summit/xl/20180502/xlC/16.1.0/bin/.orig/xlc: error: 1501-294 No input file specified. Please use -qhelp for more information.
configure:2380: $? = 249 


configure:2469: error: in `/gpfs/alpinetds/csc190/scratch/sarat/e3sm_scratch/fc5_ibm/bld/ibm/spectrum-mpi/nodebug/nothreads/mct':
configure:2471: error: C compiler cannot create executables
```
I must have missed something obvious here, any pointers to debug this are greatly appreciated.

Thanks @rzurob and @rljacob for looking into this. I tracked it down to LDFLAGS which led to missing '-lxlfmath -lxlf90_r' libraries while linking. I contacted OLCF to understand what changed in the environment.

I tried to isolate the issue by compiling a simple C file. I can see the .so.1 files for both libraries in /sw/summit/xl/20180502/lib which is added to LD_LIBRARY_PATH when loading the module. Perhaps, they need to be symlinked to .so.

Details:
```
$ xlc -o test test.c  -lxlfmath -lxlf90_r -lxlopt -lxl -lxlsmp
/usr/bin/ld: cannot find -lxlfmath
/usr/bin/ld: cannot find -lxlf90_r
/usr/bin/sha1sum: test: No such file or directory

$ ls /sw/summit/xl/20180502/lib
libibmc++.so.1  libxlcuf.so.1  libxlf90.so.1  libxlf90_r.so.1  
libibmc++libxlf90_t.so.1  libxlfmath.so.1  libxlfpmt4.so.1  
libibmc++libxlfpmt8.so.1  libxlomp_ser.so.1  libxlsmp.so.1  omptf  omptl  
libibmc++profiled

$ env | grep LD_LIBRARY
LD_LIBRARY_PATH=/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20171006/linux-rhel7-ppc64le/xl-20180502/spectrum-mpi-10.2.0.0-20180508-riohv7qa3m5koyk4u5s7lqz5drkyljr2/lib:/sw/summit/xl/20180502/xlsmp/5.1.0/lib:/sw/summit/xl/20180502/xlmass/9.1.0/lib:/sw/summit/xl/20180502/xlC/16.1.0/lib:/sw/summit/xl/20180502/xlf/16.1.0/lib:/sw/summit/xl/20180502/lib:/opt/ibm/spectrumcomputing/lsf/10.1/linux3.10-glibc2.17-ppc64le-csm/lib
```

--- a/cime/config/e3sm/machines/config_compilers.xml
+++ b/cime/config/e3sm/machines/config_compilers.xml
@@ -1072,7 +1072,7 @@ for mct, etc.
   <PIO_FILESYSTEM_HINTS>gpfs</PIO_FILESYSTEM_HINTS>
   <ADD_FFLAGS> -qzerosize -qfree=f90 -qxlf2003=polymorphic</ADD_FFLAGS>
   <ADD_FFLAGS_NOOPT> -O0 -g -qfree=f90  </ADD_FFLAGS_NOOPT>
-  <ADD_LDFLAGS>-lxlfmath -lxlf90_r -lxlopt -lxl -lxlsmp -L$(NETCDF_C_PATH)/lib -lnetcdf -L$(NETCDF_FORTRAN_PATH)/lib -lnetcdff -L$(ESSL_PATH)/lib64 -lessl -L$(NETLIB_LAPACK_PATH)/lib -llapack</ADD_LDFLAGS>
+  <ADD_LDFLAGS>-lxlopt -lxl -lxlsmp -L$(NETCDF_C_PATH)/lib -lnetcdf -L$(NETCDF_FORTRAN_PATH)/lib -lnetcdff -L$(ESSL_PATH)/lib64 -lessl -L$(NETLIB_LAPACK_PATH)/lib -llapack</ADD_LDFLAGS>
   <ADD_LDFLAGS MPILIB="!mpi-serial"> -L$(PNETCDF_PATH)/lib -lpnetcdf -L$(HDF5_PATH)/lib -lhdf5_hl -lhdf5 </ADD_LDFLAGS>
   <ADD_FFLAGS> -qspillsize=2500 -qextname=flush </ADD_FFLAGS>
   <ADD_CPPDEFS> -DLINUX  </ADD_CPPDEFS>

--- a/cime/config/e3sm/machines/config_compilers.xml
+++ b/cime/config/e3sm/machines/config_compilers.xml
@@ -162,7 +162,7 @@ for mct, etc.
 
 
   <ADD_CPPDEFS> -DFORTRANUNDERSCORE -DNO_SHR_VMATH -DNO_R16 -DCPRPGI </ADD_CPPDEFS>
-  <CFLAGS> -gopt -Mlist  -time </CFLAGS>
+  <CFLAGS> -gopt -time </CFLAGS>
 
   <ADD_CFLAGS compile_threaded="false"> </ADD_CFLAGS>
   <ADD_FFLAGS compile_threaded="false"> </ADD_FFLAGS>

--- a/cime/config/e3sm/machines/config_compilers.xml
+++ b/cime/config/e3sm/machines/config_compilers.xml
@@ -230,13 +230,12 @@ for mct, etc.
   <!--                     * inv: invalid operands         -->
   <!--                     * divz divide by zero           -->
   <!--                     * ovf: floating point overflow   -->
-  <!--  -Mlist          => Create a listing file             -->
   <!--  -F              => leaves file.f for each preprocessed file.F file  -->
   <!--  -time           => Print execution time for each compiler step  -->
 
 
   <ADD_CPPDEFS> -DFORTRANUNDERSCORE -DNO_SHR_VMATH -DNO_R16 -DUSE_CUDA_FORTRAN -DCPRPGI </ADD_CPPDEFS>
-  <CFLAGS> -Mlist -time </CFLAGS>
+  <CFLAGS> -time </CFLAGS>
 
   <ADD_CFLAGS compile_threaded="false"> </ADD_CFLAGS>
   <ADD_FFLAGS compile_threaded="false"> </ADD_FFLAGS>
@@ -249,7 +248,7 @@ for mct, etc.
   <FREEFLAGS> -Mfree </FREEFLAGS>
   <FC_AUTO_R8> -r8 </FC_AUTO_R8>
 
-  <FFLAGS>  -i4 -Mlist  -time -Mstack_arrays -Mextend -byteswapio -Mflushz -Kieee  </FFLAGS>
+  <FFLAGS>  -i4 -time -Mstack_arrays -Mextend -byteswapio -Mflushz -Kieee  </FFLAGS>
   <ADD_FFLAGS MODEL="cam"> </ADD_FFLAGS>
   <FFLAGS_NOOPT> -O0 -g -Ktrap=fp -Mbounds -Kieee  </FFLAGS_NOOPT>
   <ADD_FFLAGS_NOOPT compile_threaded="false"> </ADD_FFLAGS_NOOPT>

--- a/cime/config/e3sm/machines/config_machines.xml
+++ b/cime/config/e3sm/machines/config_machines.xml
@@ -2922,7 +2922,7 @@
         <command name="load">pgi/18.3</command>
       </modules>
       <modules compiler="ibm">
-        <command name="load">xl/20180406-beta</command>
+        <command name="load">xl/20180502</command>
       </modules>
       <modules compiler="gnu">
         <command name="load">gcc/6.4.0</command>
@@ -2935,17 +2935,17 @@
       <!-- mpi lib settings -->
 	  <!-- Sometimes,same versions of libraries are not available for different compilers, hence the split below -->
       <modules compiler="ibm" mpilib="!mpi-serial">
-        <command name="load">spectrum-mpi/10.2.0.0-20180406</command>
+        <command name="load">spectrum-mpi/10.2.0.0-20180508</command>
         <command name="load">parallel-netcdf/1.8.0</command>
         <command name="load">hdf5/1.10.0-patch1</command>
       </modules>
-      <modules compiler="pgi*" mpilib="!mpi-serial">
-        <command name="load">spectrum-mpi/10.2.0.0-20180406</command>
+      <modules compiler="pgiacc" mpilib="!mpi-serial">
+        <command name="load">spectrum-mpi/10.2.0.0-20180508</command>
         <command name="load">parallel-netcdf/1.8.0</command>
         <command name="load">hdf5/1.10.0-patch1</command>
       </modules>
       <modules compiler="gnu" mpilib="!mpi-serial">
-        <command name="load">spectrum-mpi/10.2.0.0-20180406</command>
+        <command name="load">spectrum-mpi/10.2.0.0-20180508</command>
         <command name="load">parallel-netcdf/1.8.0</command>
         <command name="load">hdf5/1.10.0-patch1</command>
       </modules>

--- a/cime/config/e3sm/machines/config_machines.xml
+++ b/cime/config/e3sm/machines/config_machines.xml
@@ -2918,7 +2918,7 @@
 		<command name="load">netlib-lapack/3.6.1</command>
       </modules>
       <!-- List of modules elements, executing commands if compiler and mpilib condition applies -->
-      <modules compiler="pgi">
+      <modules compiler="pgi.*">
         <command name="load">pgi/18.3</command>
       </modules>
       <modules compiler="ibm">
@@ -2939,7 +2939,7 @@
         <command name="load">parallel-netcdf/1.8.0</command>
         <command name="load">hdf5/1.10.0-patch1</command>
       </modules>
-      <modules compiler="pgiacc" mpilib="!mpi-serial">
+      <modules compiler="pgi.*" mpilib="!mpi-serial">
         <command name="load">spectrum-mpi/10.2.0.0-20180508</command>
         <command name="load">parallel-netcdf/1.8.0</command>
         <command name="load">hdf5/1.10.0-patch1</command>

